menubot/index.js

```diff
@@ -88,74 +105,32 @@ export async function runTurn(playerId, tickInfo, mostRecentMatchInfo, actionQue
     case "BossRoom":
       playedGame = true;
 
-      const GameHUDStartButton = getInteractableButton(tickInfo, "GameHUDStartButton");
+      const GameHUDStartButton = await getInteractableButton("GameHUDStartButton");
       if (GameHUDStartButton && stateFlags["CheatsCancelButton"] && !stateFlags["GameHUDStartButton"]) {
-        clickButton(GameHUDStartButton.id, actionQueue);
+        rg.performAction("ClickButton", {targetId: GameHUDStartButton.id});
         stateFlags["GameHUDStartButton"] = true
       }
 
-      const CheatsCancelButton = getInteractableButton(tickInfo, "CheatsCancelButton");
+      const CheatsCancelButton = await getInteractableButton("CheatsCancelButton");
       if (CheatsCancelButton && !stateFlags["CheatsCancelButton"]) {
-        clickButton(CheatsCancelButton.id, actionQueue);
+        rg.performAction("ClickButton", {targetId: CheatsCancelButton.id});
         stateFlags["CheatsCancelButton"] = true
       }
 
       break;
     case "PostGame":
     default:
       // teardown myself
-      console.log(`Game ended, bot is Complete`)
-      botComplete = true;
+      rg.complete()
       break;
   }
 
 }
 
-function getInteractableButton(tickInfo, buttonName) {
-  const buttons = RGBot.getEntitiesOfType(tickInfo, buttonName);
-  //console.log(`Found buttons: ${JSON.stringify(buttons)}`)
-  if (buttons && buttons.length > 0) {
-    const button = buttons[0];
-    if (button.interactable) {
-      //console.log(`Returning button: ${JSON.stringify(button)}`)
-      return button;
-    }
+async function getInteractableButton(buttonName) {
+  const button = await rg.findEntity(buttonName);
```

Comment:
I feel like this should be named `waitForEntity` and have some configurable timeout value.

Reply:
In non-test use cases (where you'd use runTurn) it doesn't actually wait for that entity to exist, it simply evaluates the state at that tick like the current implementation of the menubot does